pkg/agent/controller/noderoute/node_route_controller.go

```diff
@@ -206,67 +224,194 @@ func (c *Controller) syncNodeRoute(nodeName string) error {
-
-		if !flowsAreInstalled { // then install flows
-			err = c.ofClient.InstallNodeFlows(nodeName, c.nodeConfig.GatewayConfig.MAC, peerGatewayIP, *peerPodCIDR, peerNodeIP)
-			if err != nil {
-				return fmt.Errorf("failed to install flows to Node %s: %v", nodeName, err)
-			}
-			c.installedNodes.Store(nodeName, nil)
-		}
-		// install route
-		route := &netlink.Route{
-			Dst:       peerPodCIDR,
-			Flags:     int(netlink.FLAG_ONLINK),
-			LinkIndex: c.gatewayLink.Attrs().Index,
-			Gw:        peerGatewayIP,
+			return fmt.Errorf("failed to install flows to Node %s: %v", nodeName, err)
 		}
+		c.installedNodes.Store(nodeName, nil)
+	}
+
+	// install route
+	route := &netlink.Route{
+		Dst:       peerPodCIDR,
+		Flags:     int(netlink.FLAG_ONLINK),
+		LinkIndex: c.gatewayLink.Attrs().Index,
+		Gw:        peerGatewayIP,
+	}
 
-		err = netlink.RouteAdd(route)
-		// This is likely to be caused by an agent restart and so should not happen once we
-		// handle state reconciliation on restart properly. However, it is probably better
-		// to handle this case gracefully for the time being.
-		if err == unix.EEXIST {
-			klog.Warningf("Route to Node %s already exists, replacing it", nodeName)
-			err = netlink.RouteReplace(route)
+	err = netlink.RouteAdd(route)
+	// This is likely to be caused by an agent restart and so should not happen once we
+	// handle state reconciliation on restart properly. However, it is probably better
+	// to handle this case gracefully for the time being.
+	if err == unix.EEXIST {
+		klog.Warningf("Route to Node %s already exists, replacing it", nodeName)
+		err = netlink.RouteReplace(route)
+	}
+	if err != nil {
+		return fmt.Errorf("failed to install route to Node %s with netlink: %v", nodeName, err)
+	}
+	c.installedNodes.Store(nodeName, route)
+	return nil
+}
+
+// createIPSecTunnelPort creates an IPSec tunnel port for the remote Node if the
+// tunnel does not exist, and returns the ofport number.
+func (c *Controller) createIPSecTunnelPort(nodeName string, nodeIP net.IP) (int32, error) {
+	interfaceConfig, ok := c.interfaceStore.GetNodeTunnelInterface(nodeName)
+	if ok {
+		// TODO: check if Node IP, PSK, or tunnel type changes or handle it in
+		// reconciliation.
+		if interfaceConfig.OFPort != 0 {
+			return interfaceConfig.OFPort, nil
 		}
-		if err != nil {
-			return fmt.Errorf("failed to install route to Node %s with netlink: %v", nodeName, err)
+	} else {
+		portName := util.GenerateTunnelInterfaceName(nodeName)
+		ovsExternalIDs := map[string]interface{}{ovsExternalIDNodeName: nodeName}
+		if portUUID, err := c.ovsBridgeClient.CreateTunnelPortExt(
+			portName,
+			c.tunnelType,
+			0, // ofPortRequest - let OVS allocate OFPort number.
+			nodeIP.String(),
+			c.ipsecPSK,
+			ovsExternalIDs); err != nil {
+			klog.Errorf("Failed to create OVS IPSec tunnel port for Node %s: %v", nodeName, err)
+			return 0, fmt.Errorf("failed to create IPSec tunnel port for Node %s: %v", nodeName, err)
+		} else {
```

Comment:
nit: golint and most go style guides would recommend remove else to keep the normal code path at a minimal indentation.
https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow
https://golang.org/doc/effective_go.html#if

Reply:
Here, if remove else, I need to declare error and portUUID separately. What would you recommend?

Reply:
In effective go and golang wiki, it's recommended to move the statement getting the value out of if statement:
```
f, err := os.Open(name)
if err != nil {
    return err
}
codeUsing(f)
```
It can be found in the two links I mentioned.

Reply:
There are lots of such patterns in our code (I guess the main intention is to avoid declaring variables separately). But any for these two cases, you pointed out, I am able to change them easily.

Reply:
Not too many, 4 places in normal code and 10 places in test code.  Search "if block ends with a return statement, so drop this else and outdent its block" in https://goreportcard.com/report/github.com/vmware-tanzu/antrea.

Reply:
I meant "if abc, err := f(); err != nil", not "if ... else".

Anyway I changed the code as you suggested.

Reply:
To clarify my point, If the normal path doesn't use abc and err, I think it's totally fine to declare the short variables in `if` statement.